clickhouse_sqlalchemy/drivers/base.py

```diff
@@ -559,6 +552,7 @@ class ClickHouseDialect(default.DefaultDialect):
     returns_unicode_strings = True
     description_encoding = None
     postfetch_lastrowid = False
+    cached_server_version_string = None
```

Comment:
Main use case is to specify the version from outside the SA, to skip one extra network back-and-forth.

Reply:
OK, but rountrip not a problem for another dialects like pg, mysql.

Do you mind this caching be optional? `cache_server_verion=true` in dsn.

Generally it's bad idea to fool SA with inappropriate `server_version`. `server_version` is used for enabling/disabling some [dialect features](https://github.com/zzzeek/sqlalchemy/blob/rel_1_3_13/lib/sqlalchemy/dialects/postgresql/base.py#L2466).

In this dialect it can be used for multiple `JOIN` clause rending instead of raising exception for example.

Reply:
Replaced it with `forced_server_version_string`, without the caching.